p0_ex_: [-5.52305695312,-0.2138470280762,0.144448536987]
p1_ex_: [-5.52305695312,-0.2138470280762,0.144448536987]